WebRoot/public/login.jsp

```diff
@@ -384,6 +385,29 @@ if (application.getInitParameter("offlineMode") != null) {
 </c:url>
 
 <%
+	Cookie testCookieRootPath = new Cookie("ZM_TEST", "true");
+	testCookieRootPath.setMaxAge(0);
+	testCookieRootPath.setPath("/");
+	response.addCookie(testCookieRootPath);
+
+	Cookie csrfCookieRootPath = new Cookie("ZM_LOGIN_CSRF", "");
+	csrfCookieRootPath.setMaxAge(0);
+	csrfCookieRootPath.setPath("/");
+	response.addCookie(csrfCookieRootPath);
+
+	String zimbraMailURL = (String) pageContext.getAttribute("zimbraMailURL");
```

Comment:
instead of this entire code block, can't we just put the below statements at lines no 418 and 412?
```
String zimbraMailURL = (String) pageContext.getAttribute("zimbraMailURL");
testCookieNonRootPath.setPath(zimbraMailURL);
```

Reply:
Even when zimbraMailURL is set to `/test/path/`, user can access `https://host/` without the path. As described in another comment, jsp cannot identify which cookie of a path needs to be removed / updated. So the code you suggested will not solve the issue.